Core/src/main/java/com/paypal/android/core/API.kt

```diff
@@ -1,31 +1,20 @@
 package com.paypal.android.core
 
-import android.content.Context
-import android.util.Log
-import androidx.annotation.VisibleForTesting
-import com.paypal.android.core.analytics.AnalyticsEventData
-import com.paypal.android.core.analytics.DeviceInspector
-import java.util.UUID
-
 /**
  * This class is exposed for internal PayPal use only. Do not use.
  * It is not covered by Semantic Versioning and may change or be removed at any time.
  */
 class API internal constructor(
     private val configuration: CoreConfig,
-    private val sessionID: String,
     private val http: Http,
     private val httpRequestFactory: HttpRequestFactory,
-    private val deviceInspector: DeviceInspector
 ) {
 
-    constructor(configuration: CoreConfig, context: Context) :
+    constructor(configuration: CoreConfig) :
```

Comment:
Each feature client will need to still pass along a `Context` at some point.

I think we should keep the `context` as a dependency of `API`, so that each feature client can easily just call `API.sendEvent("card.started")`, for example. That method call, will delegate to an `AnalyticsService` so none of the analytics specific logic lives in `API`. Here's how I thought of doing it on iOS (https://github.com/paypal/iOS-SDK/pull/100).

With this current approach, each feature client would need _both_ an API dependency and an AnalyticsClient dependency.